src/Hydrodynamics/Hydrodynamics.cpp

```diff
@@ -202,26 +201,25 @@ void Hydrodynamics<ndim>::CheckYBoundaryGhostParticle
   const DomainBox<ndim> &simbox)       ///< [in] Simulation domain box
 {
   if (ndim > 1) {
-    Particle<ndim>& part = GetParticlePointer(i);
 
-    if (part.r[1] + min((FLOAT) 0.0, part.v[1]*tghost) <
-        simbox.boxmin[1] + ghost_range*kernrange*part.h) {
+    if (GetParticlePointer(i).r[1] + min((FLOAT) 0.0, GetParticlePointer(i).v[1]*tghost) <
```

Comment:
I think this is ugly... Wouldn't it be easier to read if we just copied r and v instead?